source/httpproxy.go

```diff
@@ -0,0 +1,323 @@
+		endpoints = append(endpoints, endpointsForHostname(hostname, targets, ttl, providerSpecific, setIdentifier)...)
+	}
+	return endpoints, nil
+}
+
+// filterByAnnotations filters a list of configs by a given annotation selector.
+func (sc *httpProxySource) filterByAnnotations(httpProxies []*projectcontour.HTTPProxy) ([]*projectcontour.HTTPProxy, error) {
+	labelSelector, err := metav1.ParseToLabelSelector(sc.annotationFilter)
+	if err != nil {
+		return nil, err
+	}
+	selector, err := metav1.LabelSelectorAsSelector(labelSelector)
+	if err != nil {
+		return nil, err
+	}
+
+	// empty filter returns original list
+	if selector.Empty() {
+		return httpProxies, nil
+	}
+
+	filteredList := []*projectcontour.HTTPProxy{}
+
+	for _, httpProxy := range httpProxies {
+		// convert the HTTPProxy's annotations to an equivalent label selector
+		annotations := labels.Set(httpProxy.Annotations)
+
+		// include HTTPProxy if its annotations match the selector
+		if selector.Matches(annotations) {
+			filteredList = append(filteredList, httpProxy)
+		}
+	}
+
+	return filteredList, nil
+}
+
+func (sc *httpProxySource) setResourceLabel(httpProxy *projectcontour.HTTPProxy, endpoints []*endpoint.Endpoint) {
+	for _, ep := range endpoints {
+		ep.Labels[endpoint.ResourceLabelKey] = fmt.Sprintf("HTTPProxy/%s/%s", httpProxy.Namespace, httpProxy.Name)
+	}
+}
+
+// endpointsFromHTTPProxyConfig extracts the endpoints from a Contour HTTPProxy object
+func (sc *httpProxySource) endpointsFromHTTPProxy(httpProxy *projectcontour.HTTPProxy) ([]*endpoint.Endpoint, error) {
+	if httpProxy.Status.CurrentStatus != "valid" {
+		log.Warn(errors.Errorf("cannot generate endpoints for HTTPProxy with status %s", httpProxy.Status.CurrentStatus))
+		return nil, nil
+	}
+
+	var endpoints []*endpoint.Endpoint
+
+	ttl, err := getTTLFromAnnotations(httpProxy.Annotations)
+	if err != nil {
+		log.Warn(err)
+	}
+
+	targets := getTargetsFromTargetAnnotation(httpProxy.Annotations)
+
+	if len(targets) == 0 {
+		for _, lb := range httpProxy.Status.LoadBalancer.Ingress {
+			if lb.IP != "" {
+				targets = append(targets, lb.IP)
+			}
+			if lb.Hostname != "" {
+				targets = append(targets, lb.Hostname)
+			}
+		}
+	}
+
+	providerSpecific, setIdentifier := getProviderSpecificAnnotations(httpProxy.Annotations)
+
+	if virtualHost := httpProxy.Spec.VirtualHost; virtualHost != nil {
+		if fqdn := virtualHost.Fqdn; fqdn != "" {
+			endpoints = append(endpoints, endpointsForHostname(fqdn, targets, ttl, providerSpecific, setIdentifier)...)
+		}
+	}
+
+	// Skip endpoints if we do not want entries from annotations
+	if !sc.ignoreHostnameAnnotation {
+		hostnameList := getHostnamesFromAnnotations(httpProxy.Annotations)
+		for _, hostname := range hostnameList {
+			endpoints = append(endpoints, endpointsForHostname(hostname, targets, ttl, providerSpecific, setIdentifier)...)
+		}
+	}
+
+	return endpoints, nil
+}
+
+func (sc *httpProxySource) AddEventHandler(ctx context.Context, handler func()) {
```

Comment:
Why not implement this? You already have an informer initialized. you just need to use that here

Reply:
I didn't do this initially because it wasn't implemented for `IngressRoute` but I could implement it.